### PR TITLE
Cleaned up the Sales Engine, moved the methods to create Item Repo & …

### PR DESCRIPTION
**Pull request by krislitman**
…Merchant Repo to perspective classes

### CODE REVIEW

**Comment by rcasias** [comment]
That looks great Kris. That makes it much more readable!

File: lib/item_repo.rb [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+  def initialize(input)
+    make_items(input)
+  end
+
+  def make_items(input)
+    items = CSV.open(input, headers: true,
+    header_converters: :symbol)
+
+    @item_list = items.map do |item|
```
Comment (klatour324) [inline lib/item_repo.rb:14]: Do we need to have the instance variable for item list here? Or can we access the reader method like we do on line 20?

File: lib/merchant_repo.rb [hunk 2]
```diff
@@ -1,8 +1,19 @@
...
-    @merchant_list = merchant_list
+  def initialize(input)
+    make_merchants(input)
+  end
+
+  def make_merchants(input)
+    merchants = CSV.open(input, headers: true,
+    header_converters: :symbol)
+
+    @merchant_list = merchants.map do |merchant|
```
Comment (klatour324) [inline lib/merchant_repo.rb:14]: Same question here as in the item_repo.rb file. For merchant list, should it  be `merchant_list = merchants...`?

**Comment by klatour324** [comment]
Nice job Kris. I can see your point about using the instance variable for `merchant_list` and `item_list`. I think this is something we can circle back and refactor.. so let's make a note of this here to do so. Great job though with breaking out the responsibility of SalesEngine to the Item_Repo and Merchant_Repo!

**Pull request merged**